Automattic/pocket-casts-android
# Display RSS artwork in all possible places

## Description

This PR redesigns how we create request for images. Previously we had `PodcastImageLoader` with multitude of different methods and it wasn't clear when to use which one. It was also hard extend this class to support handling different artworks. I removed that class and introduced `PocketCastImageRequestFactory` with a clearer and simpler API.

Episodes shown in the discovery feature do not show RSS artwork because server does not return image URL for episodes there.

Closes #1608 

## Testing Instructions

Test the screens that are using RSS artwork and see if they respect the `Use embedded RSS artwork` appearance setting. Also smoke test the rest of the app including user files, discovery, and any other place that you can think of.

## Screenshots or Screencast 

### Android

| Episode | Up Next | Filters | 
| - | - | - |
| ![android-episode](https://github.com/Automattic/pocket-casts-android/assets/30936061/eec499b3-9108-4bac-946c-9e7915cd03a3) | ![android-up-next](https://github.com/Automattic/pocket-casts-android/assets/30936061/6fa6f189-0aec-41c3-8729-98a1368b988d) | ![android-filters](https://github.com/Automattic/pocket-casts-android/assets/30936061/eda0f459-b912-469e-a085-aafdc8bd5c4c) |

| Downloads | Bookmarks | Widget |
| - | - | - |
| ![android-downlods](https://github.com/Automattic/pocket-casts-android/assets/30936061/0b9018fc-0703-4b2d-af6e-540173f3610c) | ![android-bookmarks](https://github.com/Automattic/pocket-casts-android/assets/30936061/622a9112-a527-4f65-863c-df3f66f5c7c1) | ![android-widget](https://github.com/Automattic/pocket-casts-android/assets/30936061/5ce22faa-fdc5-4477-b0c4-3ad7dc5f5e82) |

| Player | Mini player | Media notification |
| - | - | - |
| ![android-player](https://github.com/Automattic/pocket-casts-android/assets/30936061/9f8c65a8-fbe0-4aa1-94b2-ce1814967e3d) | ![android-mini-player](https://github.com/Automattic/pocket-casts-android/assets/30936061/85905039-7ca0-4157-b8e8-663fcb54dedf) | ![android-notification-center](https://github.com/Automattic/pocket-casts-android/assets/30936061/c37ac830-5d6c-47b6-8bfa-9c63d9a53e8f) |

| History | Starred |
| - | - |
| ![history](https://github.com/Automattic/pocket-casts-android/assets/30936061/dbb16228-24bc-4bbb-bfc7-6862eefce739) | ![starred](https://github.com/Automattic/pocket-casts-android/assets/30936061/09d58816-b9b3-454f-be4c-810e5c2847c3) |

In theory notifications should use this as well but I'm not able to test it properly because our developer testing action clears the episodes which means that don't have image URLs when fetched.

### Wear OS

| Now Playing | Up Next | Filters | Downloads |
| - | - | - | - |
| ![wear-now-playing](https://github.com/Automattic/pocket-casts-android/assets/30936061/b3e6ed60-3bc8-43a7-85c7-3e7f6aef3c38) | ![wear-up-next](https://github.com/Automattic/pocket-casts-android/assets/30936061/a61d0b81-827a-4c4c-9007-d81c50e7e501) | ![wear-filters](https://github.com/Automattic/pocket-casts-android/assets/30936061/be0736df-9d58-4de0-a596-2fc7c2e2b123) | ![wear-downloads](https://github.com/Automattic/pocket-casts-android/assets/30936061/1b1c43e2-d45e-462f-b9c5-70d31644f825) |

### Automotive

No changes to the automotive codebase and it still works as before. It is a bit finicky to test because there might be some delay due to image URL availability in episode entities.

## Checklist
 
#### I have tested any UI changes...
